rpcs3/rpcs3qt/settings_dialog.cpp

```diff
@@ -973,6 +974,36 @@ settings_dialog::settings_dialog(std::shared_ptr<gui_settings> guiSettings, std:
 
 	xemu_settings->EnhanceCheckBox(ui->spuDebug, emu_settings::SPUDebug);
 	SubscribeTooltip(ui->spuDebug, json_debug["spuDebug"].toString());
+		
+	if (utils::has_rtm() && !utils::has_mpx())
+	{
+		xemu_settings->EnhanceCheckBox(ui->enableTSX, emu_settings::EnableTSX);
+		SubscribeTooltip(ui->enableTSX, json_debug["enableTSX"].toString());
+
+		// connect the toogled signal so that the stateChanged signal in EnhanceCheckBox can be prevented
+		connect(ui->enableTSX, &QCheckBox::toggled, [this](bool checked)
+		{
+			if (checked && QMessageBox::No == QMessageBox::critical(this, tr("Haswell/Broadwell TSX Warning"), tr(
```

Comment:
In order to make this universal change this to:
`if (checked && !utils::has_mpx() && QMessageBox::No == ....`